cloud/amazon/ec2_lc_find.py

```diff
@@ -0,0 +1,226 @@
+    type: string
+    sample: {"Enabled": false}
+classic_link_vpc_security_groups:
+    description: Launch Configuration classic link vpc security groups property
+    returned: when Launch Configuration was found
+    type: list
+    sample: []
+block_device_mappings:
+    description: Launch Configuration block device mappings property
+    returned: when Launch Configuration was found
+    type: list
+    sample: []
+keyname:
+    description: Launch Configuration ssh key
+    returned: when Launch Configuration was found
+    type: string
+    sample: mykey
+keyname:
+    description: Launch Configuration security groups
+    returned: when Launch Configuration was found
+    type: list
+    sample: []
+kernel_id:
+    description: Launch Configuration kernel to use
+    returned: when Launch Configuration was found
+    type: string
+    sample: ''
+ram_disk_id:
+    description: Launch Configuration ram disk property
+    returned: when Launch Configuration was found
+    type: string
+    sample: ''
+associate_public_address:
+    description: Assign public address or not
+    returned: when Launch Configuration was found
+    type: boolean
+    sample: True
+...
+'''
+
+
+def find_launch_configs(client, module):
+    name_regex = module.params.get('name_regex')
+    sort_order = module.params.get('sort_order')
+    offset = module.params.get('limit')
+
+    paginator = client.get_paginator('describe_launch_configurations')
+
+    response_iterator = paginator.paginate(
+        PaginationConfig={
+            'MaxItems': 1000,
+            'PageSize': 100
+        }
+    )
+
+    for response in response_iterator:
+        response['LaunchConfigurations'] = filter(lambda lc: re.compile(name_regex).match(lc['LaunchConfigurationName']),
+                                                  response['LaunchConfigurations'])
+
+    results = []
+    for lc in response['LaunchConfigurations']:
+        data = {
+            'name': lc['LaunchConfigurationName'],
+            'arn': lc['LaunchConfigurationARN'],
+            'created_time': lc['CreatedTime'],
+            'user_data': lc['UserData'],
+            'instance_type': lc['InstanceType'],
+            'image_id': lc['ImageId'],
+            'ebs_optimized': lc['EbsOptimized'],
+            'instance_monitoring': lc['InstanceMonitoring'],
+            'classic_link_vpc_security_groups': lc['ClassicLinkVPCSecurityGroups'],
+            'block_device_mappings': lc['BlockDeviceMappings'],
+            'keyname': lc['KeyName'],
+            'security_groups': lc['SecurityGroups'],
+            'kernel_id': lc['KernelId'],
+            'ram_disk_id': lc['RamdiskId'],
+            'associate_public_address': lc['AssociatePublicIpAddress'],
+        }
+
+        results.append(data)
+
+    results.sort(key=lambda e: e['name'], reverse=(sort_order == 'descending'))
+
+    try:
+        if offset:
+            results = results[:int(offset)]
+    except TypeError:
```

Comment:
see argspec comment below- you can nuke this check if you just tell us what the type should be